reactive/kubernetes_master.py

```diff
@@ -2063,37 +2068,77 @@ def configure_apiserver():
             kubectl('delete', 'service', 'kubernetes')
+            actual_service_ip = _wait_for_svc_ip()
+            if not actual_service_ip:
+                # we might need another restart to get the service recreated
+                hookenv.log("service-cidr expansion: Timed out waiting for "
+                            "the service to return; restarting API server")
+                clear_flag('kubernetes-master.apiserver.configured')
+                return
+            if actual_service_ip != expected_service_ip:
+                raise ValueError('Unexpected service IP: {} != {}'.format(
+                    actual_service_ip, expected_service_ip))
+
+        # Restart the cdk-addons
+        # Get deployments/daemonsets/statefulsets
+        hookenv.log("service-cidr expansion: Restart the cdk-addons")
+        output = kubectl(
+            'get', 'daemonset,deployment,statefulset',
+            '-o', 'json',
+            '--all-namespaces',
+            '-l', 'cdk-restart-on-ca-change=true'
+        ).decode('UTF-8')
+        deployments = json.loads(output)['items']
 
-            # Restart the cdk-addons
-            # Get deployments/daemonsets/statefulsets
-            hookenv.log("service-cidr expansion: Restart the cdk-addons")
-            output = kubectl(
-                'get', 'daemonset,deployment,statefulset',
-                '-o', 'json',
-                '--all-namespaces',
-                '-l', 'cdk-restart-on-ca-change=true'
-            ).decode('UTF-8')
-            deployments = json.loads(output)['items']
-
-            # Now restart the addons
-            for deployment in deployments:
-                kind = deployment['kind']
-                namespace = deployment['metadata']['namespace']
-                name = deployment['metadata']['name']
-                hookenv.log('Restarting addon: {0} {1} {2}'.format(kind,
-                                                                   namespace,
-                                                                   name))
-                kubectl(
-                    'rollout', 'restart', kind + '/' + name,
-                    '-n', namespace
-                )
-        except Exception:
-            hookenv.log("service-cidr-expansion: k8s services not yet started")
-
-    set_flag('kubernetes-master.apiserver.configured')
+        # Now restart the addons
+        for deployment in deployments:
+            kind = deployment['kind']
+            namespace = deployment['metadata']['namespace']
+            name = deployment['metadata']['name']
+            hookenv.log('Restarting addon: {0} {1} {2}'.format(kind,
+                                                               namespace,
+                                                               name))
+            kubectl(
+                'rollout', 'restart', kind + '/' + name,
+                '-n', namespace
+            )
+    except CalledProcessError:
```

Comment:
`hookenv.log(traceback.format_exc())` please

Reply:
Per IRC, the kubectl calls are already logged here: https://github.com/charmed-kubernetes/layer-kubernetes-common/blob/be49e03a77e9abee648aba39e707ea39d8982427/lib/charms/layer/kubernetes_common.py#L220

So that should be good enough to figure out what's going on when this fails.